lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -88,21 +88,11 @@ def self.included(mod)
       mod.config :port, :obsolete => "Please use the 'hosts' setting instead. Hosts entries can be in 'host:port' format."
 
       # This plugin uses the bulk index API for improved indexing performance.
-      # In Logstashes >= 2.2 this setting defines the maximum sized bulk request Logstash will make
+      # This setting defines the maximum sized bulk request Logstash will make
       # You you may want to increase this to be in line with your pipeline's batch size.
       # If you specify a number larger than the batch size of your pipeline it will have no effect,
       # save for the case where a filter increases the size of an inflight batch by outputting
       # events.
-      #
-      # In Logstashes <= 2.1 this plugin uses its own internal buffer of events.
-      # This config option sets that size. In these older logstashes this size may
-      # have a significant impact on heap usage, whereas in 2.2+ it will never increase it.
-      # To make efficient bulk API calls, we will buffer a certain number of
-      # events before flushing that out to Elasticsearch. This setting
-      # controls how many events will be buffered before sending a batch
-      # of events. Increasing the `flush_size` has an effect on Logstash's heap size.
-      # Remember to also increase the heap size using `LS_HEAP_SIZE` if you are sending big documents
-      # or have increased the `flush_size` to a higher value.
       mod.config :flush_size, :validate => :number, :default => 500
```

Comment:
I think its ok to mark it `deprecated` if we have to keep this setting around for just one use case (the 120MB event). IMO this is an edge case. We should discourage folks from using `flush_size` because you have the `pipeline-batch-size`. Having to tweak event batches in 2 places is just confusing. 

Open to discussing this via zoom with a larger group